### PR TITLE
feat: Add ability to define custom timeout for fargate profiles

### PR DESCRIPTION
**Pull request by IvanDechovsky**
Add ability to define custom timeout for create/delete operations for fargate profiles

https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1610

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -96,6 +96,11 @@ module "eks" {
         }
       ]
 
+      timeouts = {
+        create = "15"
```
Comment (daroga0002) [inline examples/complete/main.tf:100]:
```
Error: Error parsing "create" timeout: time: missing unit in duration "15"
```

File: examples/fargate/main.tf [hunk 2]
```diff
@@ -51,6 +51,12 @@ module "eks" {
       # Using specific subnets instead of the ones configured in EKS (`subnets` and `fargate_subnets`)
       subnets = [local.vpc.private_subnets[1]]
 
+      # Use custom defaults for create/delete operations
+      timeouts = {
+        create = "15"
```
Comment (daroga0002) [inline examples/fargate/main.tf:56]:
```
Error: Error parsing "create" timeout: time: missing unit in duration "15"
```

File: examples/fargate/main.tf [hunk 3]
```diff
@@ -110,6 +116,12 @@ module "fargate_profile_existing_cluster" {
         }
       ]
 
+      # Use custom defaults for create/delete operations
+      timeouts = {
+        create = "15"
```
Comment (daroga0002) [inline examples/fargate/main.tf:121]:
```
Error: Error parsing "create" timeout: time: missing unit in duration "15"
```

File: modules/fargate/README.md [hunk 4]
```diff
@@ -10,25 +10,26 @@ Helper submodule to create and manage resources related to `aws_eks_fargate_prof
...
-|------|-------------|------|---------|:--------:|
-| name | Fargate profile name | `string` | Auto generated in the following format `[cluster_name]-fargate-[fargate_profile_map_key]`| no |
-| selectors | A list of Kubernetes selectors. See examples/fargate/main.tf for example format. | <pre>list(map({<br>namespace = string<br>labels = map(string)<br>}))</pre>| `[]` | no |
-| subnets | List of subnet IDs. Will replace the root module subnets. | `list(string)` | `var.subnets` | no |
-| tags | Key-value map of resource tags. Will be merged with root module tags. | `map(string)` | `var.tags` | no |
+| Name      | Description                                                                      | Type                                                                       | Default                                                                                   | Required |
+| --------- | -------------------------------------------------------------------------------- | -------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------- | :------: |
```
Comment (daroga0002) [inline modules/fargate/README.md:14]: does those changes where done by you or by some automation?

File: modules/fargate/README.md [hunk 5]
```diff
@@ -37,36 +38,36 @@ No modules.
...
 ## Outputs
 
-| Name | Description |
-|------|-------------|
-| <a name="output_aws_auth_roles"></a> [aws\_auth\_roles](#output\_aws\_auth\_roles) | Roles for use in aws-auth ConfigMap |
-| <a name="output_fargate_profile_arns"></a> [fargate\_profile\_arns](#output\_fargate\_profile\_arns) | Amazon Resource Name (ARN) of the EKS Fargate Profiles. |
-| <a name="output_fargate_profile_ids"></a> [fargate\_profile\_ids](#output\_fargate\_profile\_ids) | EKS Cluster name and EKS Fargate Profile names separated by a colon (:). |
-| <a name="output_iam_role_arn"></a> [iam\_role\_arn](#output\_iam\_role\_arn) | IAM role ARN for EKS Fargate pods |
-| <a name="output_iam_role_name"></a> [iam\_role\_name](#output\_iam\_role\_name) | IAM role name for EKS Fargate pods |
+| Name                                                                                                 | Description                                                              |
```
Comment (daroga0002) [inline modules/fargate/README.md:66]: what has done those changes?

this part of file should be autogenerated via pre commit hooks

File: modules/fargate/main.tf [hunk 6]
```diff
@@ -63,5 +63,10 @@ resource "aws_eks_fargate_profile" "this" {
     }
   }
 
+  timeouts {
+    create = lookup(each.value["timeouts"], "create", 10)
```
Comment (daroga0002) [inline modules/fargate/main.tf:67]: instead assuming some values put failover to `null` value which will use provider defaults

Comment (daroga0002) [reply modules/fargate/main.tf:67]:
```suggestion
    create = lookup(each.value["timeouts"], "create", null)
```

File: modules/fargate/main.tf [hunk 7]
```diff
@@ -63,5 +63,10 @@ resource "aws_eks_fargate_profile" "this" {
...
+    delete = lookup(each.value["timeouts"], "delete", 10)
```
Comment (daroga0002) [inline modules/fargate/main.tf:68]:
```suggestion
    delete = lookup(each.value["timeouts"], "delete", null)
```

**Review by daroga0002 — CHANGES_REQUESTED** [review]
This change seems be not tested, please test and fix a comments


Additionally I don`t know does changes in README.md where done by you or any automatic ?? (we use precommit hooks to generate some part of docs using terraform-docs)

File: variables.tf [hunk 8]
```diff
@@ -387,6 +387,18 @@ variable "fargate_profiles" {
   default     = {}
 }
 
+variable "fargate_profiles_create_timeout" {
```
Comment (daroga0002) [inline variables.tf:390]: We dont want add additionall variables into submodule, it should follow previously used pattern. I dont know why you changed this as I given you direct suggestion how to make default null value

**Review by daroga0002 — CHANGES_REQUESTED** [review]
We should same pattern across timeouts (was used previously in this PR, I dont know why you changed)

**Comment by IvanDechovsky** [comment]
@daroga0002 Will you have some time to review it this week perhaps?

**Comment by IvanDechovsky** [comment]
Any chance this can be looked at please?

**Comment by daroga0002** [comment]
I have this on my list, will do this till end of week

File: examples/fargate/main.tf [hunk 9]
```diff
@@ -154,6 +166,12 @@ module "fargate_profile_existing_cluster" {
         Owner     = "profile2"
         submodule = "true"
       }
+
+      # Set custom timeout for create/delete operation on fargate profiles
+      timeouts = {
+        create = "20m"
```
Comment (daroga0002) [inline examples/fargate/main.tf:172]: lets remove `create` just for test case when we have just one subargument

File: examples/fargate/main.tf [hunk 10]
```diff
@@ -89,6 +95,12 @@ module "eks" {
       tags = {
         Owner = "secondary"
       }
+
+      # Set custom timeout for create/delete operation on fargate profiles
```
Comment (daroga0002) [inline examples/fargate/main.tf:99]: remove lines 99-103 for test case when we dont define any timeout arguments

**Review by daroga0002 — CHANGES_REQUESTED** [review]
Please add into file 
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/modules/fargate/README.md

in line 17 following content:
```
| timeouts | A map of timeouts for create/delete operations. | `map(string)` | Provider default behavior | no |
```

**Comment by daroga0002** [comment]
@IvanDechovsky thank you for your contribution, 3 comments and after fixing we will be able to merge it

**Comment by IvanDechovsky** [comment]
@daroga0002 Thank you for taking a look. Updated!

File: modules/fargate/README.md [hunk 11]
```diff
@@ -15,6 +15,7 @@ See example code in `examples/fargate`.
 | name | Fargate profile name | `string` | Auto generated in the following format `[cluster_name]-fargate-[fargate_profile_map_key]`| no |
 | selectors | A list of Kubernetes selectors. See examples/fargate/main.tf for example format. | <pre>list(map({<br>namespace = string<br>labels = map(string)<br>}))</pre>| `[]` | no |
 | subnets | List of subnet IDs. Will replace the root module subnets. | `list(string)` | `var.subnets` | no |
+| timeouts | A map of timeouts for create/delete operations. | `map(string)` | Provider default behavior | no |
```
Comment (daroga0002) [inline modules/fargate/README.md:18]:
```suggestion
| timeouts | A map of timeouts for create/delete operations. | `map(string)` | Provider default behaviour | no |
```

**Review by daroga0002 — APPROVED** [review]
One typo in `behaviour` to fix and we can merge.

@antonbabenko  this can be merged when typo will be fixed (maybe you can accept commit on suggestion?)

File: modules/fargate/README.md (hunk 11 above)
Comment (antonbabenko) [reply modules/fargate/README.md:18]: US vs UK :) Let's leave the English language to native-language speakers (definitely, not me and you)

**Comment by antonbabenko** [comment]
This PR is included in version 17.24.0 :tada:

**Pull request merged**